File: backend/app/models/ticket.py
from pydantic import BaseModel
from typing import Optional

class TicketBase(BaseModel):
    EventID: int
    Name: str
    Price: float
    OnePer: int
    QuantityTotal: int

class TicketCreate(TicketBase):
    pass

class TicketUpdate(BaseModel):
    Name: Optional[str] = None
    Price: Optional[float] = None
    OnePer: Optional[int] = None
    QuantityTotal: Optional[int] = None
    QuantitySold: Optional[int] = None

class TicketResponse(TicketBase):
    TicketID: int
    QuantitySold: int

    class Config:
        from_attributes = True

File: event_manager_app/lib/domain/entities/ticket.dart
class TicketEntity {
  final int ticketId;
  final int eventId;
  final String name;
  final double price;
  final int onePer;
  final int quantityTotal;
  final int quantitySold;

  TicketEntity({
    required this.ticketId,
    required this.eventId,
    required this.name,
    required this.price,
    required this.onePer,
    required this.quantityTotal,
    required this.quantitySold,
  });

  int get availableQuantity => quantityTotal - quantitySold;
  bool get isAvailable => availableQuantity > 0;
}
